### PR TITLE
Write git revision into last_deploy file

### PR DESCRIPTION
**Pull request by ChrisBr**
to know which revision gets installed. 
We also display now the revision on the /about route

### CODE REVIEW

File: dist/obs-server.spec [hunk 1]
```diff
@@ -341,7 +340,7 @@ export DESTDIR=$RPM_BUILD_ROOT
 
 export OBS_VERSION="%{version}"
 DESTDIR=%{buildroot} make install FILLUPDIR=%{_fillupdir}
-
+sed -n -e '/commit: /,$p' %{_sourcedir}/open-build-service.obsinfo | tr -d "commit: " > %{buildroot}/srv/www/obs/api/last_deploy
```
Comment (ChrisBr) [inline dist/obs-server.spec]: @eduardoj maybe you have a more elegant way to achieve this:
I have obs-server.obsinfo file with the following content:

```
name: open-build-service
version: 2.10~alpha.20180507T142315.30c3301a1
mtime: 1525695795
commit: 30c3301a1d25b8bad1505a5e16ff287289934b6c
```

and I want to write the commit hash into the ``last_deploy`` file.

**Comment by adrianschroeter** [comment]
On Dienstag, 8. Mai 2018, 10:33:26 CEST wrote Christian Bruckmayer:
> ChrisBr commented on this pull request.
> 
> 
> 
> > @@ -341,7 +340,7 @@ export DESTDIR=$RPM_BUILD_ROOT
>  
>  export OBS_VERSION="%{version}"
>  DESTDIR=%{buildroot} make install FILLUPDIR=%{_fillupdir}
> -
> +sed -n -e '/commit: /,$p' %{_sourcedir}/open-build-service.obsinfo | tr -d "commit: " > %{buildroot}/srv/www/obs/api/last_deploy
> 
> @eduardoj maybe you have a more elegant way to achieve this:
> I have obs-server.obsinfo file with the following content:
> 
> ```
> name: open-build-service
> version: 2.10~alpha.20180507T142315.30c3301a1
> mtime: 1525695795
> commit: 30c3301a1d25b8bad1505a5e16ff287289934b6c
> ```
> 
> and I want to write the commit hash into the ``last_deploy`` file.

why not just

echo %{version} > %{buildroot}/srv/www/obs/api/last_deploy

?

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

File: dist/obs-server.spec (hunk 1 above)
Comment (hennevogel) [reply dist/obs-server.spec]: `awk -F: 'END{print $2}' file`

**Comment by ChrisBr** [comment]
> echo %{version} > %{buildroot}/srv/www/obs/api/last_deploy

because it is not the commit hash, it is sth like ``2.10~alpha.20180505T002410.07ad05e20``. Also for releases it does not contain any hash.

**Comment by adrianschroeter** [comment]
On Dienstag, 8. Mai 2018, 11:04:38 CEST wrote Henne Vogelsang:
> hennevogel commented on this pull request.
> 
> 
> 
> > @@ -341,7 +340,7 @@ export DESTDIR=$RPM_BUILD_ROOT
>  
>  export OBS_VERSION="%{version}"
>  DESTDIR=%{buildroot} make install FILLUPDIR=%{_fillupdir}
> -
> +sed -n -e '/commit: /,$p' %{_sourcedir}/open-build-service.obsinfo | tr -d "commit: " > %{buildroot}/srv/www/obs/api/last_deploy
> 
> `awk -F: 'END{print $2}' file`

nah, please don't use the obsinfo file, but the %version macro.

The file may not be there if we decide to build from a tar ball...

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

**Comment by adrianschroeter** [comment]
On Dienstag, 8. Mai 2018, 11:05:23 CEST wrote Christian Bruckmayer:
> > echo %{version} > %{buildroot}/srv/www/obs/api/last_deploy
> 
> because it is not the commit hash, it is sth like ``2.10~alpha.20180505T002410.07ad05e20``. Also for releases it does not contain any hash.

.07ad06e20  is the hash, no?


-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

**Comment by hennevogel** [comment]
> The file may not be there if we decide to build from a tar ball...

If we do that we can also decide to use `%version` then...

**Comment by ChrisBr** [comment]
> .07ad06e20  is the hash, no?

part of it, yes! But as said, it does not exist e.g. for releases.

**Comment by adrianschroeter** [comment]
On Dienstag, 8. Mai 2018, 11:10:53 CEST wrote Christian Bruckmayer:
> > .07ad06e20  is the hash, no?
> 
> part of it, yes! But as said, it does not exist e.g. for releases.

who cares, it should be tagged ;)

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

File: dist/obs-server.spec (hunk 1 above)
Comment (eduardoj) [reply dist/obs-server.spec]: @ChrisBr, this works for me:

`sed -n -e 's/commit: \(.\+\)/\1/p' %{_sourcedir}/open-build-service.obsinfo > %{buildroot}/srv/www/obs/api/last_deploy`

**Comment by ChrisBr** [comment]
>> part of it, yes! But as said, it does not exist e.g. for releases.

> who cares, it should be tagged ;)

Sure, but what do we write into the last_deploy file in this case? Nothing? Unknown? The code will be much more complicated because I would need to check first if it is unstable or not and then parse the revision out of the version or write sth else to the file ... 
I prefer what we do now ...

**Comment by ChrisBr** [comment]
Discussed with @adrianschroeter offline and the consent is to check for the file existence which I do now! Merging ...

**Pull request merged**